src/features/formData/FormDataWrite.tsx

```diff
@@ -224,11 +235,11 @@ function FormDataEffects() {
   // to trigger when the user is typing, which is not what we want.
   useEffect(
     () => () => {
-      if (hasUnsavedChangesRef.current) {
+      if (hasUnsavedChangesRef.current && !isSavingNow()) {
         performSave();
       }
     },
-    [hasUnsavedChangesRef, performSave],
+    [hasUnsavedChangesRef, isSavingNow, performSave],
   );
```

Comment:
Does this need to check `isSavingNow()`? Is this not already checked by the `hasUnsavedChangesRef`? Maybe I dont understand exactly what causes this to update, but this should only happen when unmounting the component?

Reply:
You're probably right. The problem I've had with this state is that its way too often out of date, and will cause multiple saves. Maybe I've resorted to check things twice, and it's quite possible I forgot to remove this again when it didn't turn out to be fully working. I think the best solution so far has been to make `useHasUnsavedChangesRef` into an anonymous class that actually checks if there are unsaved changes at the point when you read `ref.current`. But I guess that could just as easily be a function call now, so I'll look into this some more when tests are stable again. :pray:

Reply:
Looks like that went well! :raised_hands: